src/balance.ts

```diff
@@ -1,72 +1,62 @@
-import * as errors from "./errors";
 import { BigNumber } from "bignumber.js";
+import { ESDTToken } from "./esdtToken";
+import { ErrInvalidArgument } from "./errors";
+import { Egld } from "./balanceBuilder";
 
 /**
- * The base used for toString methods to avoid exponential notation
+ * The number of decimals handled when working with EGLD or ESDT values.
  */
-const BASE_10 = 10;
+const DEFAULT_BIGNUMBER_DECIMAL_PLACES = 18;
 
-/**
- * The number of decimals handled when working with EGLD values.
- */
-const DENOMINATION = 18;
-
-/**
- * One EGLD, in its big-integer form (as a string).
- */
-const OneEGLDString = "1000000000000000000";
-
-const EGLDTicker = "EGLD";
 
-BigNumber.set({ DECIMAL_PLACES: DENOMINATION, ROUNDING_MODE: 1 });
+BigNumber.set({ DECIMAL_PLACES: DEFAULT_BIGNUMBER_DECIMAL_PLACES, ROUNDING_MODE: 1 });
```

Comment:
Because this sets a global variable in the BigNumber library and it affects division operations.
https://mikemcl.github.io/bignumber.js/#decimal-places

It should be a default value large enough to cover most tokens (hence why it stays 18), but it's independent from the number of decimals a token uses (hence the rename).
The actual number of decimals a token uses is set in the `this.token.decimals` variable - and it's only that variable which matters for displaying denominated values.